### PR TITLE
Controlnet ssd 1b support

### PR DESCRIPTION
**Pull request by MarkoKostiv**
# What does this PR do?

- This PR adds UNetMidBlock2D option for the mid-block in ControlNet, so you can train it with [segmind/SSD-1B](https://huggingface.co/segmind/SSD-1B) as the base model.
- It also allows you to redefine the conditioning number of channels when you initialize weights from base model Unet for training

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
CC @patrickvonplaten and @sayakpaul

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
This looks good to me and I am in favor of this because SSD-1B is quite a powerful model. 

Have you tried training any ControlNets on top of SSD-1B? 

@patrickvonplaten thoughts?

**Comment by MarkoKostiv** [comment]
> Have you tried training any ControlNets on top of SSD-1B?

Yes, I made sure ssd-1b works with ControlNet and converges, but no checkpoints yet to share with the community.

**Comment by sayakpaul** [comment]
Cc: @Gothos for visibility.

**Comment by MarkoKostiv** [comment]
The `code quality check` failure seems to be unrelated to the current changes.
image

**Comment by patrickvonplaten** [comment]
I'm in favor of supporting this!

**Comment by patrickvonplaten** [comment]
Can we make sure to add some tests here?

**Comment by MarkoKostiv** [comment]
> Can we make sure to add some tests here?

I added some tests based on the SDXL pipeline (since it's highly compatible with SSD-1b) and overridden the differences.

**Review by sayakpaul — APPROVED** [review]
Looks very cool! Will merge after the CI is green.

**Comment by MarkoKostiv** [comment]
> @MarkoKostiv thanks for your contributions once again. Could you look into the failing test so that we can merge the PR.

@sayakpaul All green! Happy to contribute!

File: tests/pipelines/controlnet/test_controlnet_sdxl.py [hunk 1]
```diff
@@ -817,3 +818,162 @@ def test_depth(self):
         original_image = images[0, -3:, -3:, -1].flatten()
         expected_image = np.array([0.4399, 0.5112, 0.5478, 0.4314, 0.472, 0.4823, 0.4647, 0.4957, 0.4853])
         assert np.allclose(original_image, expected_image, atol=1e-04)
+
+
+class StableDiffusionSSD1BControlNetPipelineFastTests(StableDiffusionXLControlNetPipelineFastTests):
```
Comment (sayakpaul) [inline tests/pipelines/controlnet/test_controlnet_sdxl.py:823]: Wondering if this should go into a separate testing script (potentially named `test_controlnet_ssd.py`)? WDYT @patrickvonplaten?

**Review by sayakpaul — APPROVED** [review]
LGTM! Thank you!

File: tests/pipelines/controlnet/test_controlnet_sdxl.py (hunk 1 above)
Comment (patrickvonplaten) [reply tests/pipelines/controlnet/test_controlnet_sdxl.py:823]: Think it's ok to leave here

**Pull request merged**